python/lsst/drp/tasks/wcsfit.py

```diff
@@ -0,0 +1,1145 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+import numpy as np
+import astropy.time
+import astropy.units as u
+import astropy.coordinates
+import yaml
+import wcsfit
+import pandas as pd
+import astshim
+
+import lsst.geom
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.sphgeom
+import lsst.afw.table
+import lsst.afw.geom as afwgeom
+from lsst.meas.algorithms import (LoadReferenceObjectsConfig, ReferenceObjectLoader,
+                                  ReferenceSourceSelectorTask)
+from lsst.meas.algorithms.sourceSelector import sourceSelectorRegistry
+
+__all__ = ["WCSFitConnections", "WCSFitConfig", "WCSFitTask"]
+
+
+def lookup_visit_refcats(datasetType, registry, quantumDataId, collections):
+    """Lookup function that finds all refcats for all visits that overlap a
+    tract, rather than just the refcats that directly overlap the tract.
+    Borrowed from jointcal.
+
+    Parameters
+    ----------
+    datasetType : `lsst.daf.butler.DatasetType`
+        Type of dataset being searched for.
+    registry : `lsst.daf.butler.Registry`
+        Data repository registry to search.
+    quantumDataId : `lsst.daf.butler.DataCoordinate`
+        Data ID of the quantum; expected to be something we can use as a
+        constraint to query for overlapping visits.
+    collections : `Iterable` [ `str` ]
+        Collections to search.
+    Returns
+    -------
+    refs : `Iterator` [ `lsst.daf.butler.DatasetRef` ]
+        Iterator over refcat references.
+    """
+    refs = set()
+    # Use .expanded() on the query methods below because we need data IDs with
+    # regions, both in the outer loop over visits (queryDatasets will expand
+    # any data ID we give it, but doing it up-front in bulk is much more
+    # efficient) and in the data IDs of the DatasetRefs this function yields
+    # (because the RefCatLoader relies on them to do some of its own
+    # filtering).
+    for visit_data_id in set(registry.queryDataIds("visit", dataId=quantumDataId).expanded()):
+        refs.update(
+            registry.queryDatasets(
+                datasetType,
+                collections=collections,
+                dataId=visit_data_id,
+                findFirst=True,
+            ).expanded()
+        )
+    yield from refs
+
+
+def make_ref_covariance_matrix(refCat, inputUnit=u.radian, outputCoordUnit=u.marcsec, outputPMUnit=u.marcsec):
```

Comment:
These utility functions aren't in the exported `__all__`.  I assume this is intentional, and these are just utility functions used by the task?  If so, I wonder if it would make sense to rename them all with a leading underscore to make it super clear these aren't supposed to be public.